sklearn/tree/tree.py

```diff
@@ -108,6 +108,14 @@ def __init__(self,
         self.class_weight = class_weight
         self.presort = presort
 
+    def get_max_depth(self):
```

Comment:
I'd agree. That's how I had it, then I double checked and realized for some reason @amueller seems confident they should be getters (that's how I felt reading the comments on #7613). I'd say `max_depth_` can definitely be a `@property` since it's just returning a value. But I'm not sure about `n_leaves_`, since it's of `O(n)`, `n` being the number of nodes.

Reply:
I think we usually avoid ``@properties`` because they hide computation and possibly side-effects from the users.